### PR TITLE
LPS-181116 Implement display of the list of fields in FDS View Fields tab

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-181116 Implement display of the list of fields in FDS View Fields tab](https://issues.liferay.com/browse/LPS-181116)
- [Mockups](https://www.figma.com/file/VTOLTBFe99y9kA7AY3TcNR/lps-165276-design-app-to-manage-datasets?node-id=1212-129807&t=O4mTLVakfJIDVQrk-0)

### What is the goal of this PR?

The main goal of this PR is to fetch list of fields for a given REST application and schema. 

Technical notes:
- We are making a OpenAPI REST application request and parsing response for given schema properties.
- Properties are in different format for Objects and other REST applications. In either case, there are some properties that do not match fields. Unfortunately, there is no explicit marker that a property is a field. Logic that we are applying is probably the ugliest part of this PR (see use of `extensions` property and removal of `x-class-name`). Hopefully, solution in this PR is temporary, and we get some better indication from origin later on. /cc @4lejandrito @gabrielwas
- We're doing a couple of side-quests:
  - renaming remaining types and interfaces
  - moving API URL definitions from backend to frontend
- We are preparing persistence for FDS Fields. It is not yet used, will be used in the next story.

/cc @ugeortiz

### What does it look like?

https://user-images.githubusercontent.com/5435511/232493024-4693b6bf-efc5-422b-b8ba-e5e863c19dcf.mp4

### Steps to reproduce
1. Add `feature.flag.LPS-164563=true` to `portal-ext.properties`. This enables FDS Views Manager.
1. Add `feature.flag.LPS-161364=true` to `portal-ext.properties`. This enables Objects relationships nested fields, used by FDS Views Manager.
1. If you previously opened Datasets page, delete `FDS Entry` and `FDS View` on Global Menu > Control Panel > Object > Objects
1. Open Global Menu > Control Panel > Object > Datasets

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+										</ClayInput.GroupItem>
+									</ClayInput.Group>
+								</ManagementToolbar.Item>
+							</ManagementToolbar.ItemList>
+						</ManagementToolbar.Container>
+
+						<div className="fields pb-2 pt-2">
+							{fields.map(({name, selected, visible}) => (
+								<>
+									{visible && (
```
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx]: Force-pushed, before anybody notices :ninja:

**Comment by mateusralv** [comment]
ci:test:relevant

**Review by kresimir-coko — APPROVED** [review]
💪

**Comment by mateusralv** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
This PR will need to be updated after https://github.com/liferay-frontend/liferay-portal/pull/3243 is merged, to fix conflicts and renaming updates. No need for CI testing until then.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java [hunk 2]
```diff
@@ -181,6 +181,46 @@ private synchronized void _generate(
...
+				userId, false, false,
+				LocalizedMapUtil.getLocalizedMap("FDS Field"), "FDSField",
+				"300", null, LocalizedMapUtil.getLocalizedMap("FDS Fields"),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+				Arrays.asList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						_language.get(locale, "column-label"), "label", false),
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:196]: I'm fine with this for the moment. In the future this model (which shall probably be named FieldMapping) will need some changes to accommodate other non-table view modes.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java [hunk 3]
```diff
@@ -181,6 +181,46 @@ private synchronized void _generate(
...
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						_language.get(locale, "name"), "name", true),
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:200]: Is this the name of the field we're mapping?

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java [hunk 4]
```diff
@@ -181,6 +181,46 @@ private synchronized void _generate(
...
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						_language.get(locale, "type"), "type", true),
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:204]: We're storing the type of the field taken from the openapi schema right?

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java (hunk 3 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:200]: Yes.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java (hunk 4 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:204]: Yes. This, along with `name`, should be read-only. `label` and `renderer` should be set by admin. I assumed in this PR that they are not mandatory, i.e. empty on save. We may want to set some defaults, copying `name` into `label` and setting `renderer` to `default`. cc @kresimir-coko

**Comment by kresimir-coko** [comment]
I'm having some trouble POST-ing this, what data do I need to send as the body of the fetch?

The `field` as it is represented in the frontend looks like this:
```json
{
    "name": "name",
    "selected": true,
    "type": "string",
    "visible": true
}
```

We have 2 POST endpoints, `/` and `/batch`. Which one should I use?

It's allowing me to POST just one field to the `/` endpoint.
For more than one selected field I need to use the `/batch` endpoint, where I get a 202 Accepted (not processed) response, specifically:
```json
{
  "className": "com.liferay.object.rest.dto.v1_0.ObjectEntry",
  "contentType": "JSON",
  "errorMessage": "",
  "executeStatus": "INITIAL",
  "externalReferenceCode": "3fe9c526-2690-c32a-ea44-1d1e5b4748be",
  "failedItems": [],
  "id": 4,
  "importStrategy": "ON_ERROR_FAIL",
  "operation": "CREATE",
  "processedItemsCount": 0,
  "startTime": "2023-04-19T09:42:32Z",
  "totalItemsCount": 0
}
```

In the API explorer it was complaining about a `sortable` property missing from the field.

I've got the POST to somewhat work, but there are some bugs preventing it from working 100%

**Pull request closed without merging**